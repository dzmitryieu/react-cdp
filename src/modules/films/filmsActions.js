import axios from 'axios';
import { cleanSearchField } from '../searchFilms/searchFilmsActions';

export const FETCH_FILMS_REQUEST = 'FETCH_FILMS_REQUEST';
export const FETCH_FILMS_SUCCESS = 'FETCH_FILMS_SUCCESS';
export const FETCH_FILMS_ERROR = 'FETCH_FILMS_ERROR';
export const SORT_FILMS = 'SORT_FILMS';

export const urlFilms = 'https://reactjs-cdp.herokuapp.com/movies';

export const receiveFilmsSuccess = payload => ({
  type: FETCH_FILMS_SUCCESS,
  payload,
});

export const receiveFilmsRequest = () => ({
  type: FETCH_FILMS_REQUEST,
});

export const receiveFilmsError = payload => ({
  type: FETCH_FILMS_ERROR,
  payload,
});

export const sortFilms = payload => ({
  type: SORT_FILMS,
  payload,
});

export function fetchFilms(url) {
  return (dispatch) => {
    dispatch(receiveFilmsRequest());
    return axios.get(url)
      .then((response) => {
        dispatch(receiveFilmsSuccess({ films: response.data.data, url }));
      })
      .catch(error => dispatch(receiveFilmsError(error.message)));
  };
}

export function fetchAllFilms() {
  return (dispatch) => {
    dispatch(cleanSearchField());
    return dispatch(fetchFilms(urlFilms));
  };
}

export function fetchFilmsBySearch(word, searchBy) {
  return (dispatch) => {
    dispatch(fetchFilms(`${urlFilms}?search=${word}&searchBy=${searchBy}`));
  };
}
