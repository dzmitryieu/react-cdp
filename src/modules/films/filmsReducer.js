import {
  FETCH_FILMS_REQUEST,
  FETCH_FILMS_SUCCESS,
  FETCH_FILMS_ERROR,
  SORT_FILMS,
} from './filmsActions';
import initialState from './index';
import sort from '../../utils/sort';

const filmsReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_FILMS_SUCCESS:
      return {
        ...state,
        isFetchingFilms: false,
        isFetchedFilms: true,
        allFilms: action.payload.films,
        url: action.payload.url,
        errorFilms: '',
      };
    case FETCH_FILMS_REQUEST:
      return {
        ...state,
        isFetchingFilms: true,
        isFetchedFilms: false,
      };
    case FETCH_FILMS_ERROR:
      return {
        ...state,
        errorFilms: action.payload,
        isFetchingFilms: false,
        isFetchedFilms: false,
        allFilms: [],
      };
    case SORT_FILMS:
      return {
        ...state,
        allFilms: [].concat(sort(state.allFilms, action.payload.direction, action.payload.key)),
      };
    default:
      return {
        ...state,
      };
  }
};

export default filmsReducer;
