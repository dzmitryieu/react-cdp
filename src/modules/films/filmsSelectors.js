export const getFilms = state => state.persistedStore.allFilms;
export const getIsFetchingFilms = state => state.persistedStore.isFetchingFilms;
export const getIsFetchedFilms = state => state.persistedStore.isFetchedFilms;
export const getErrorFilms = state => state.persistedStore.errorFilms;
export const getMainFilm = state => state.persistedStore.mainFilm;
export const isShowSearchPage = state => state.persistedStore.showSearchPage;
