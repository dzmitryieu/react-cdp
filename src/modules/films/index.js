const initialState = {
  allFilms: [],
  errorFilms: {},
  isFetchingFilms: false,
  isFetchedFilms: false,
  url: '',
  mainFilm: {},
};

export default initialState;
