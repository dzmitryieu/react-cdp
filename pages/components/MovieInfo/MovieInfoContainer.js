import { connect } from 'react-redux';
import MovieInfo from './MovieInfo';
import { getFilms } from '../../../src/modules/films/filmsSelectors';


const mapStateToProps = state => ({
  films: getFilms(state),
});

export default connect(mapStateToProps)(MovieInfo);
