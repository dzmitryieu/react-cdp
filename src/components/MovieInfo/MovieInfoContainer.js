import { connect } from 'react-redux';
import MovieInfo from './MovieInfo';
import { fetchFilmByRoute } from '../../modules/film/filmActions';
import { getFilms, getIsFetchingFilms } from '../../modules/films/filmsSelectors';


const mapStateToProps = state => ({
  films: getFilms(state),
  isFetchingFilms: getIsFetchingFilms(state),
});

const mapDispatchToProps = {
  fetchFilmByRoute,
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieInfo);
