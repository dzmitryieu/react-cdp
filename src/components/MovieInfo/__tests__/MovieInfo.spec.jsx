import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import MovieInfoContainer from '../index';
import films from '../../../modules/mocks/getFilmsMock.json';


const allFilms = films.data;
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  films: { allFilms },
});

describe('MovieInfo Snapshot', () => {
  test('render with store', () => {
    const component = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <MovieInfoContainer />
        </Router>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
