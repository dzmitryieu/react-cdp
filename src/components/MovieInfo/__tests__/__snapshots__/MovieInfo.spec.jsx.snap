// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo Snapshot render 1`] = `
<div
  className="list"
>
  <Film
    film={
      Object {
        "genres": Array [
          "Drama",
          "Comedy",
        ],
        "id": 353463654,
        "overview": "Super movie sdfsfs",
        "poster_path": "../../assets/images/poster.jpg",
        "release_date": "1984-12-03",
        "runtime": 150,
        "tagline": "Super movie",
        "title": "Captain Marvel",
      }
    }
    onClick={[Function]}
  />
  <Film
    film={
      Object {
        "genres": Array [
          "Drama",
          "Comedy",
        ],
        "id": 4676898,
        "overview": "Super movie sdfsfs",
        "poster_path": "../../assets/images/poster.jpg",
        "release_date": "1984-12-03",
        "runtime": 150,
        "tagline": "Super movie",
        "title": "Captain Marvel",
      }
    }
    onClick={[Function]}
  />
  <Film
    film={
      Object {
        "genres": Array [
          "Drama",
          "Comedy",
        ],
        "id": 23141515,
        "overview": "Super movie sdfsfs",
        "poster_path": "../../assets/images/poster.jpg",
        "release_date": "1984-12-03",
        "runtime": 150,
        "tagline": "Super movie",
        "title": "Captain Marvel",
      }
    }
    onClick={[Function]}
  />
  <Film
    film={
      Object {
        "genres": Array [
          "Drama",
          "Comedy",
        ],
        "id": 9068756,
        "overview": "Super movie sdfsfs",
        "poster_path": "../../assets/images/poster.jpg",
        "release_date": "1984-12-03",
        "runtime": 150,
        "tagline": "Super movie",
        "title": "Captain Marvel",
      }
    }
    onClick={[Function]}
  />
  <Film
    film={
      Object {
        "genres": Array [
          "Drama",
          "Comedy",
        ],
        "id": 133453456,
        "overview": "Super movie sdfsfs",
        "poster_path": "../../assets/images/poster.jpg",
        "release_date": "1984-12-03",
        "runtime": 150,
        "tagline": "Super movie",
        "title": "Captain Marvel",
      }
    }
    onClick={[Function]}
  />
</div>
`;

exports[`MovieInfo Snapshot render without films in store 1`] = `
<div
  className="list"
/>
`;

exports[`MovieInfo Snapshot with store render 1`] = `
<div
  className="list"
>
  <div
    className="container"
    id="film"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="presentation"
  >
    <div
      className="wrapper"
    >
      <div
        className="poster"
        style={
          Object {
            "backgroundImage": "url(../../assets/images/poster.jpg)",
          }
        }
      />
      <div
        className="container"
      >
        <div
          className="film-description"
        >
          <div
            className="titleWrapper"
          >
            <div
              className="name"
            >
              Captain Marvel
            </div>
            <div
              className="year"
            >
              1984
            </div>
          </div>
          <div
            className="info"
          >
            <div
              className="genres"
            >
              <span
                className="genre"
              >
                Drama
              </span>
              <span
                className="genre"
              >
                Comedy
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
    id="film"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="presentation"
  >
    <div
      className="wrapper"
    >
      <div
        className="poster"
        style={
          Object {
            "backgroundImage": "url(../../assets/images/poster.jpg)",
          }
        }
      />
      <div
        className="container"
      >
        <div
          className="film-description"
        >
          <div
            className="titleWrapper"
          >
            <div
              className="name"
            >
              Captain Marvel
            </div>
            <div
              className="year"
            >
              1984
            </div>
          </div>
          <div
            className="info"
          >
            <div
              className="genres"
            >
              <span
                className="genre"
              >
                Drama
              </span>
              <span
                className="genre"
              >
                Comedy
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
    id="film"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="presentation"
  >
    <div
      className="wrapper"
    >
      <div
        className="poster"
        style={
          Object {
            "backgroundImage": "url(../../assets/images/poster.jpg)",
          }
        }
      />
      <div
        className="container"
      >
        <div
          className="film-description"
        >
          <div
            className="titleWrapper"
          >
            <div
              className="name"
            >
              Captain Marvel
            </div>
            <div
              className="year"
            >
              1984
            </div>
          </div>
          <div
            className="info"
          >
            <div
              className="genres"
            >
              <span
                className="genre"
              >
                Drama
              </span>
              <span
                className="genre"
              >
                Comedy
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
    id="film"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="presentation"
  >
    <div
      className="wrapper"
    >
      <div
        className="poster"
        style={
          Object {
            "backgroundImage": "url(../../assets/images/poster.jpg)",
          }
        }
      />
      <div
        className="container"
      >
        <div
          className="film-description"
        >
          <div
            className="titleWrapper"
          >
            <div
              className="name"
            >
              Captain Marvel
            </div>
            <div
              className="year"
            >
              1984
            </div>
          </div>
          <div
            className="info"
          >
            <div
              className="genres"
            >
              <span
                className="genre"
              >
                Drama
              </span>
              <span
                className="genre"
              >
                Comedy
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
    id="film"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="presentation"
  >
    <div
      className="wrapper"
    >
      <div
        className="poster"
        style={
          Object {
            "backgroundImage": "url(../../assets/images/poster.jpg)",
          }
        }
      />
      <div
        className="container"
      >
        <div
          className="film-description"
        >
          <div
            className="titleWrapper"
          >
            <div
              className="name"
            >
              Captain Marvel
            </div>
            <div
              className="year"
            >
              1984
            </div>
          </div>
          <div
            className="info"
          >
            <div
              className="genres"
            >
              <span
                className="genre"
              >
                Drama
              </span>
              <span
                className="genre"
              >
                Comedy
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
